### PR TITLE
CB-13979: More consistency for config.xml lookups

### PR DESCRIPTION
**Pull request by dpogue**
Fix as proposed in https://issues.apache.org/jira/browse/CB-13979

I'd like to try to get a unit test written to confirm this fix before merging.

### CODE REVIEW

File: spec/fixtures/test-config.xml [hunk 1]
```diff
@@ -82,6 +82,7 @@
     <preference name="orientation" value="portrait" />
     <icon id="icon" src="icon.png" />
     <icon height="255" id="logo" src="logo.png" width="255" />
+    <platform name="android"></platform>
```
Comment (stevengill) [inline spec/fixtures/test-config.xml]: Why the extra android here?

Comment (dpogue) [reply spec/fixtures/test-config.xml]: That is what causes the bug described in CB-13979, although I agree it's a weird scenario.

If I add the empty platform tag without the rest of these changes, existing tests fail.
With these changes, they pass even when the empty platform tag is present.

Comment (dpogue) [reply spec/fixtures/test-config.xml]: All tests pass with the empty platform tag removed.

With these changes, all tests also pass with the empty platform tag added.

Comment (stevengill) [reply spec/fixtures/test-config.xml]: Sounds good. Feel free to merge if you feel good about it!

**Pull request merged**